sycl/include/CL/sycl/stream.hpp

```diff
@@ -104,6 +107,19 @@ class stream {
 
   bool operator!=(const stream &LHS) const;
 
+  ~stream() {
+    // Flush data to global buffer in stream destruction if flush buffer is not
+    // empty. This could be necessary if user hasn't flushed data himself and
+    // kernel execution is finished
+    // NOTE: In the current implementation user should explicitly flush data on
+    // the host device. Data is not flushed automatically after kernel execution
+    // because of the missing feature in scheduler.
+    auto Offset = getOffsetPtr(FlushBufs, FlushBufferSize, WIOffset);
+    if ((Offset != nullptr) && *Offset)
```

Comment:
is this if statement redundant? It seems this is checked already inside flushBuffer.

Reply:
Thanks, you are right, fixed.